src/Build/Evaluation/ItemSpec.cs

```diff
@@ -244,22 +257,52 @@ internal abstract class ItemFragment
         /// </summary>
         protected string ProjectDirectory { get; }
 
-        private readonly Lazy<FileSpecMatcherTester> _fileMatcher;
-        private readonly Lazy<IMSBuildGlob> _msbuildGlob;
-        protected virtual IMSBuildGlob MsBuildGlob => _msbuildGlob.Value;
+        private bool _fileMatcherInitialized;
+        private FileSpecMatcherTester _fileMatcher;
+
+        // not a Lazy to reduce memory
+        private FileSpecMatcherTester FileMatcher
+        {
+            get
+            {
+                if (_fileMatcherInitialized)
+                {
+                    return _fileMatcher;
+                }
+
+                _fileMatcher = CreateFileSpecMatcher();
```

Comment:
This could be created multiple times is that a problem?

Reply:
Thought about that as well, but after looking at how it's called, it should be single threaded. Evaluation and Project apis are single threaded. Then even if they were multi threaded, the Itemspecs themselves would be thread local. If however, in the future, the itemspecs end up getting shared between threads then yes, this will be a problem.

Do you think it's worth prematurely making it thread safe again?

Reply:
No, just wanted to make sure it was deliberate.